Refactor to include recommended serializations

aphillips opened the pull request:
This is a massive refactoring and so is best viewed via preview rather than by staring at the diff.

In this PR I brought in the text I created in localizable-manifests to describe serializations.
I then merged best practices and associated text already in this document into appropriate sections.
I split the best practices section from the introduction.
I retitled sections in best practices for accuracy.

~~In general, there is very little new/changed prose here.~~

Further edits include:
* I added a new "core" best practice, similar to my comments to e.g. VC.
* Moved additional best practices to the end of the BP section
* Retitled the section about "tech specific solutions" to just "JSON-LD" because that's all that's in there
* Moved the WebIDL BP to single-string section
* Created a U+200F image for the character reference I added (replacing a casual mention of RLM) including putting it into the echidna manifest


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/string-meta/pull/81.html" title="Last updated on Jan 19, 2024, 7:46 PM UTC (8e0384a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/string-meta/81/f2e02c1...aphillips:8e0384a.html" title="Last updated on Jan 19, 2024, 7:46 PM UTC (8e0384a)">Diff</a>

aphillips committed:
Refactor to include recommended serializations

aphillips committed:
Further reorganization plus new core bp

aphillips committed:
Reorder single-string section items

aphillips committed:
Add U+200F image
- includes echidna manifest!

aphillips committed:
Fix missing word.

aphillips committed:
Remove stray mentions of manifest

aphillips committed:
Put the id in the right place

aphillips committed:
Clean up syntactic content
- Add more glossary references
- Distinguish some cases where we don't mean syntactic content

aphillips committed:
Merge branch 'w3c:gh-pages' into gh-pages

aphillips committed:
Tighten text around non-linguistic fields

aphillips committed:
Merge branch 'gh-pages' of https://github.com/aphillips/string-meta into gh-pages

aphillips committed:
Fix typos, clean up language in document defaults

xfq commented on index.html:
```diff
@@ -45,7 +45,7 @@
 <body>
 <div id="abstract">
-  <p>This document describes the best practices for identifying language and base direction for strings used on the Web.</p>
+  <p>This document describes the best practices for identifying language and paragraph direction for strings used on the Web.</p>
```
Why was this changed to "paragraph direction"?

xfq replied:
If we decide to use "paragraph direction", we might want to mention it in https://www.w3.org/International/articles/lang-bidi-use-cases/index.en.html#definitions too.

aphillips committed:
Address @kleinsin's comment in 2024-01-04 call

Wrote a new "note" in the section about legacy string values based
on the conversation in above mentioned telecon.

aphillips committed:
Add 'and as a last resort'

This was discussed in the 2024-01-11 teleconference.

aphillips committed:
Highlight the use cases document

aphillips commented:
Per 2024-01-11 call, merging this. Will make a separate PR regarding base vs. paragraph direction
